diff --git a/app/Kconfig b/app/Kconfig
index 6b4e3509..dbddc633 100644
--- a/app/Kconfig
+++ b/app/Kconfig
@@ -543,6 +543,7 @@ config ZMK_LOGGING_MINIMAL
 if !ZMK_LOGGING_MINIMAL
 
 config ZMK_LOG_LEVEL
+    int "ZMK Log Level"
     default 4
 
 endif
diff --git a/app/src/pointing/input_split.c b/app/src/pointing/input_split.c
index b08d7ac7..00d41e94 100644
--- a/app/src/pointing/input_split.c
+++ b/app/src/pointing/input_split.c
@@ -56,8 +56,15 @@ int zmk_input_split_report_peripheral_event(uint8_t reg, uint8_t type, uint16_t
                  "Peripheral input splits need an `input` property set");                          \
     void split_input_handler_##n(struct input_event *evt) {                                        \
         for (size_t i = 0; i < ARRAY_SIZE(processors_##n); i++) {                                  \
-            zmk_input_processor_handle_event(processors_##n[i].dev, evt, processors_##n[i].param1, \
-                                             processors_##n[i].param2, NULL);                      \
+            int ret =zmk_input_processor_handle_event(processors_##n[i].dev, evt, processors_##n[i].param1,\
+                processors_##n[i].param2, NULL);                      \
+            if (ret < 0) {                                                                         \
+                LOG_ERR("Error %d from input processor %s", ret,                                   \
+                   processors_##n[i].dev->name);                                                  \
+                return;                                                                            \
+            } else if (ret == ZMK_INPUT_PROC_STOP) {                                               \
+                return;                                                                            \
+            }                     \
         }                                                                                          \
         struct zmk_split_transport_peripheral_event ev = {                                         \
             .type = ZMK_SPLIT_TRANSPORT_PERIPHERAL_EVENT_TYPE_INPUT_EVENT,                         \
@@ -68,7 +75,11 @@ int zmk_input_split_report_peripheral_event(uint8_t reg, uint8_t type, uint16_t
                          .value = evt->value,                                                      \
                          .sync = evt->sync,                                                        \
                      }}};                                                                          \
-        zmk_split_peripheral_report_event(&ev);                                                    \
+        int ret = zmk_split_peripheral_report_event(&ev);                                                      \
+        if (ret < 0) {                                                                            \
+            LOG_ERR("Error %d from zmk_split_bt_report_input", ret);                               \
+            return;                                                                                \
+        }                                                 \
     }                                                                                              \
     INPUT_CALLBACK_DEFINE(DEVICE_DT_GET(DT_INST_PHANDLE(n, device)), split_input_handler_##n);
 
diff --git a/app/west.yml b/app/west.yml
index ce0785e3..08d4d465 100644
--- a/app/west.yml
+++ b/app/west.yml
@@ -7,7 +7,7 @@ manifest:
   projects:
     - name: zephyr
       remote: zmkfirmware
-      revision: v3.5.0+zmk-fixes
+      revision: 0562599c90ae1b1b14337c3fdcb13744725f21c8
       clone-depth: 1
       import:
         name-blocklist:
